include/exadg/convection_diffusion/user_interface/parameters.h

```diff
@@ -233,6 +234,10 @@ class Parameters
   // polynomial degree of shape functions
   unsigned int degree;
 
+  // enable adaptive mesh refinement
+  bool                       enable_adaptivity;
+  AdaptiveMeshRefinementData amr_data;
```

Comment:
I don't remember exactly: Did we discuss whether these parameters should belong to `GridData` or whether they should not? One idea would be that inside `GridData` one could easily make certain checks/asserts that will then hold for all the modules, e.g. that `enable_adaptivity` is currently not possible with `Simplex`. I am not sure if e.g. this particular condition is implemented somewhere in the code currently?

Reply:
To be more precise: I think it still makes sense to keep all the parameters currently in `AdaptiveMeshRefinementData` in a separate struct related to AMR, in order to not mix these parameters with other grid parameters. One might want to define different types of AMR-data, so it could make sense to keep `amr_data` outside `grid_data`. Having `enable_adaptivity` inside `grid_data`, however, would easily allow the assert mentioned above. what do you think?

Reply:
I think I mean something different here. For every new module, we would have to add this boolean parameter, while we would not have to add it in case it would be part of `GridData`. 

The check between `enable_adaptivity` and `simplex` inside `GridData` is more *encapsulated* than in some free function (which one does not have to use; users/developers might write different functions potentially and would then have to think of such asserts again).

Maybe let's discuss this in person.